ereolen_support/ereolen_support.form.inc

```diff
@@ -0,0 +1,182 @@
+    '#required' => variable_get('ereolen_support_description_required')
+  ];
+
+  $form['ereolen_support_date'] = [
+    '#type' => 'date_popup',
+    '#title' => t('Dato'),
+    '#date_format' => array_pop(system_get_date_formats('ding_material_lists_date'))['format'],
+    '#date_label_position' => 'none',
+    '#description' => variable_get('ereolen_support_date_description'),
+    '#required' => variable_get('ereolen_support_date_required')
+  ];
+
+  $form['ereolen_support_book_title'] = [
+    '#type' => 'textfield',
+    '#title' => t('What book does the problem concern?'),
+    '#description' => variable_get('ereolen_support_book_title_description'),
+    '#required' => variable_get('ereolen_support_book_title_required')
+  ];
+
+  $form['ereolen_support_device'] = [
+    '#type' => 'select',
+    '#title' => t('What device did you use?'),
+    '#options' => _ereolen_support_create_select_list('ereolen_support_device'),
+    '#description' => variable_get('ereolen_support_device_description'),
+    '#required' => variable_get('ereolen_support_device_required')
+  ];
+
+  $form['ereolen_support_model'] = [
+    '#type' => 'textfield',
+    '#title' => t('On what device model did you experience the problem?'),
+    '#description' => variable_get('ereolen_support_model_description'),
+    '#required' => variable_get('ereolen_support_model_required')
+  ];
+
+  $form['ereolen_support_operating_system'] = [
+    '#type' => 'textfield',
+    '#title' => t('On what operating system did you experience the problem?'),
+    '#description' => variable_get('ereolen_support_operating_system_description'),
+    '#required' => variable_get('ereolen_support_model_operating_system')
+  ];
+
+  $form['submit_button'] = [
+    '#type' => 'submit',
+    '#value' => t('Send'),
+  ];
+
+  return $form;
+}
+
+/**
+ * Add a submit handler to the support form.
+ *
+ * @param $form
+ *   The form to submit.
+ * @param $form_state
+ *   The state of the form.
+ *
+ * @throws \Exception
+ */
+function ereolen_support_form_submit($form, &$form_state) {
+
+  global $user;
+
+  // Check for and create user in Jira
+  $jiraUser = _ereolen_support_search_jira_user($user, $form_state);
+  if (!$jiraUser) {
+    $jiraUser = _ereolen_support_create_jira_user($user, $form_state);
+  }
+
+  // @todo switch between jira and email This functionality is awaiting customer.
+
+  // Sometimes send to publizon mail.
+  $language = $user->uid ? user_preferred_language($user) : language_default();
+  $headers = [
+    'MIME-Version' => '1.0',
+    'Content-Type' => 'text/html; charset=UTF-8; format=flowed; delsp=yes',
+    'Content-Transfer-Encoding' => '8Bit',
+    'X-Mailer' => 'Drupal',
+  ];
+
+  $mail_params = [
+    'headers' => $headers,
+    'form_state' => $form_state
+  ];
+
+  //drupal_mail('ereolen_support', 'ereolen_support_publizon_support', variable_get('publizon_support_email'), $language, $mail_params, $from = variable_get('site_mail'));
```

Comment:
?

Reply:
This is still awaiting eReolen decision on when to send mail to publizon rather than creating an issue.